azurerm/resource_arm_application_gateway_test.go

```diff
@@ -0,0 +1,1265 @@
+  name                = "vnet"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  address_space       = ["10.254.0.0/16"]
+  location            = "${azurerm_resource_group.test.location}"
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = "subnet"
+  resource_group_name  = "${azurerm_resource_group.test.name}"
+  virtual_network_name = "${azurerm_virtual_network.test.name}"
+  address_prefix       = "10.254.0.0/24"
+}
+
+resource "azurerm_public_ip" "test" {
+  name                         = "public-ip"
+  location                     = "${azurerm_resource_group.test.location}"
+  resource_group_name          = "${azurerm_resource_group.test.name}"
+  public_ip_address_allocation = "dynamic"
+}
+
+resource "azurerm_application_gateway" "test" {
+  name                = "acctestgw-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  sku {
+    name     = "Standard_Medium"
+    tier     = "Standard"
+    capacity = 1
+  }
+
+  disabled_ssl_protocols = [
+    "TLSv1_0",
+  ]
+
+  gateway_ip_configuration {
+    # id = computed
+    name      = "gw-ip-config1"
+    subnet_id = "${azurerm_subnet.test.id}"
+  }
+
+  frontend_ip_configuration {
+    # id = computed
+    name                 = "ip-config-public"
+    public_ip_address_id = "${azurerm_public_ip.test.id}"
+  }
+
+  frontend_ip_configuration {
+    # id = computed
+    name      = "ip-config-private"
+    subnet_id = "${azurerm_subnet.test.id}"
+
+    # private_ip_address = computed
+    private_ip_address_allocation = "Dynamic"
+  }
+
+  frontend_port {
+    # id = computed
+    name = "port-8080"
+    port = 8080
+  }
+
+  backend_address_pool {
+    # id = computed
+    name = "pool-1"
+
+    fqdn_list = [
+      "terraform.io",
+    ]
+  }
+
+  backend_http_settings {
+    # id = computed
+    name                  = "backend-http-1"
+    port                  = 8010
+    protocol              = "Https"
+    cookie_based_affinity = "Enabled"
+    request_timeout       = 30
+
+    # probe_id = computed
+    probe_name = "probe-1"
+  }
+
+  http_listener {
+    # id = computed
+    name = "listener-1"
+
+    # frontend_ip_configuration_id = computed
+    frontend_ip_configuration_name = "ip-config-public"
+
+    # frontend_ip_port_id = computed
+    frontend_port_name = "port-8080"
+    protocol           = "Http"
+  }
+
+  http_listener {
+    name                           = "listener-2"
+    frontend_ip_configuration_name = "ip-config-public"
+    frontend_port_name             = "port-8080"
+    protocol                       = "Https"
+
+    # ssl_certificate_id = computed
+    ssl_certificate_name = "ssl-1"
+    host_name            = "terraform.io"
+    require_sni          = true
+  }
+
+  probe {
+    # id = computed
+    name                = "probe-1"
+    protocol            = "Https"
+    path                = "/test"
+    host                = "azure.com"
```

Comment:
out of interest does this need to match [this list](https://github.com/terraform-providers/terraform-provider-azurerm/pull/120/files#diff-d035020bd0e4dd3f83f1c7a41c6cbf16R384)?